third_party/java/arcs/build_defs/internal/tools.oss.bzl

```diff
@@ -15,21 +19,73 @@ def arcs_manifest_parse_test(name, srcs, deps = []):
         if not src.endswith(".arcs"):
             fail("src must be an .arcs manifest file, found %s" % src)
 
-    manifest_name = name + "_sources"
+    test_args = " ".join(["--src $(location %s)" % src for src in srcs])
 
-    arcs_manifest(
-        name = manifest_name,
+    sigh_command(
+        name = name,
         srcs = srcs,
         deps = deps,
+        sigh_cmd = "run manifestChecker " + test_args,
+        progress_message = "Checking Arcs manifest",
+        execute = False,
     )
 
-    test_args = " ".join(["--src $(location %s)" % src for src in srcs])
+# buildifier: disable=function-docstring
+def arcs_tool_recipe2plan(name, srcs, outs, deps, generate_proto = False, recipe = None):
+    sigh_cmd = "recipe2plan --quiet --outdir $(dirname {OUT}) --outfile $(basename {OUT})"
+    if generate_proto:
+        sigh_cmd += " --format proto"
+    if recipe:
+        sigh_cmd += " --recipe " + recipe
+    sigh_cmd += " {SRC}"
 
     sigh_command(
         name = name,
         srcs = srcs,
-        deps = deps + [":" + manifest_name],
-        sigh_cmd = "run manifestChecker " + test_args,
-        progress_message = "Checking Arcs manifest",
-        execute = False,
+        outs = outs,
+        progress_message = "Generating Kotlin Plans",
+        sigh_cmd = sigh_cmd,
+        deps = deps,
+    )
+
+# buildifier: disable=function-docstring
+def arcs_tool_manifest2json(name, srcs, outs, deps):
+    sigh_command(
+        name = name,
+        srcs = srcs,
+        outs = outs,
+        progress_message = "Serializing manifest",
+        sigh_cmd = "manifest2json --outdir $(dirname {OUT}) --outfile $(basename {OUT}) {SRC}",
+        deps = deps,
+    )
+
+# buildifier: disable=function-docstring
+def arcs_tool_manifest2proto(name, srcs, outs, deps):
+    sigh_command(
+        name = name,
+        srcs = srcs,
+        outs = outs,
+        deps = deps,
+        progress_message = "Serializing manifest",
+        sigh_cmd = "manifest2proto --quiet --outdir $(dirname {OUT}) --outfile $(basename {OUT}) {SRC}",
+    )
+
+# buildifier: disable=function-docstring
+def arcs_tool_schema2wasm(name, srcs, outs, deps, language_name, language_flag, wasm, test_harness):
+    # TODO: generated header guard should contain whole workspace-relative
```

Comment:
I see this TODO was moved with the sigh command. 

Is it being tracked anywhere or by anyone?

Reply:
I have no idea what this TODO means, sorry. I just kept it because it was there.